### PR TITLE
Allow to set the curves used

### PR DESCRIPTION
**Pull request by normanmaurer**
Motivation:

Sometimes it is useful to restrict the curves that are used.

Modifications:

Expose SSL_CTX_set1_curves_list via JNI

Result:

Be able to configure the curves that are used for a context

### CODE REVIEW

File: openssl-dynamic/src/main/java/io/netty/internal/tcnative/SSLContext.java [hunk 1]
```diff
@@ -694,4 +694,31 @@ public static void setPrivateKeyMethod(long ctx, AsyncSSLPrivateKeyMethod method
...
+        if (curves.length == 0) {
+            throw new IllegalArgumentException();
+        }
+        StringBuilder sb = new StringBuilder();
+        for (String curve: curves) {
+            sb.append(curve);
+            // Curves are separated by : as explained in the manpage.
+            sb.append(':');
+        }
+        sb.setLength(sb.length() - 1);
```
Comment (chrisvest) [inline openssl-dynamic/src/main/java/io/netty/internal/tcnative/SSLContext.java:719]: `StringJoiner` (new in Java 8) can simplify this.

**Review by chrisvest — APPROVED** [review]
Had a suggestion, but leaving it up to you if it's worth it or not.

File: openssl-dynamic/src/main/java/io/netty/internal/tcnative/SSLContext.java (hunk 1 above)
Comment (normanmaurer) [reply openssl-dynamic/src/main/java/io/netty/internal/tcnative/SSLContext.java:719]: I think it is not worth it to depend on java8 just for that

**Review by chrisvest — APPROVED** [review]
Looks like an ordeal to get old openssl versions supported

**Pull request merged**